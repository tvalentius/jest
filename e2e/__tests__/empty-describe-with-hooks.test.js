/**
 * Copyright (c) 2018-present, Facebook, Inc. All rights reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow
 */

'use strict';

const path = require('path');
const runJest = require('../runJest');
const {extractSummary} = require('../Utils');
const dir = path.resolve(__dirname, '../empty-describe-with-hooks');
const ConditionalTest = require('../../scripts/ConditionalTest');

ConditionalTest.skipSuiteOnJasmine();

test('hook in empty describe', () => {
  const result = runJest(dir, ['hook-in-empty-describe.test.js']);
  expect(result.status).toBe(1);
  expect(extractSummary(result.stderr)).toMatchSnapshot();
});

test('hook in describe with skipped test', () => {
  const result = runJest(dir, ['hook-in-describe-with-skipped-test.test.js']);
  expect(result.status).toBe(0);
  expect(extractSummary(result.stderr)).toMatchSnapshot();
});

test('hook in empty nested describe', () => {
  const result = runJest(dir, ['hook-in-empty-nested-describe.test.js']);
  expect(result.status).toBe(1);
  expect(extractSummary(result.stderr)).toMatchSnapshot();
});

test('multiple hooks in empty describe', () => {
  const result = runJest(dir, ['multiple-hooks-in-empty-describe.test.js']);
  expect(result.status).toBe(1);
  expect(extractSummary(result.stderr)).toMatchSnapshot();
});
