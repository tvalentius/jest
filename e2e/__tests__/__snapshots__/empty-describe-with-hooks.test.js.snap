// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hook in describe with skipped test 1`] = `
Object {
  "rest": "",
  "summary": "Test Suites: 1 skipped, 0 of 1 total
Tests:       1 skipped, 1 total
Snapshots:   0 total
Time:        <<REPLACED>>
Ran all test suites matching /hook-in-describe-with-skipped-test.test.js/i.",
}
`;

exports[`hook in empty describe 1`] = `
Object {
  "rest": "FAIL __tests__/hook-in-empty-describe.test.js


  ● Test suite failed to run

    Invalid: beforeEach() may not be used in a describe block containing no tests.

       7 | 
       8 | describe('a block', () => {
    >  9 |   beforeEach(() => {});
         |   ^
      10 | });
      11 | 
      12 | describe('another block with tests', () => {

      at __tests__/hook-in-empty-describe.test.js:9:3
      at __tests__/hook-in-empty-describe.test.js:8:1

",
  "summary": "Test Suites: 1 failed, 1 total
Tests:       1 passed, 1 total
Snapshots:   0 total
Time:        <<REPLACED>>
Ran all test suites matching /hook-in-empty-describe.test.js/i.
",
}
`;

exports[`hook in empty nested describe 1`] = `
Object {
  "rest": "FAIL __tests__/hook-in-empty-nested-describe.test.js


  ● Test suite failed to run

    Invalid: beforeEach() may not be used in a describe block containing no tests.

       7 | 
       8 | describe('a block', () => {
    >  9 |   beforeEach(() => {});
         |   ^
      10 |   describe('another block', () => {});
      11 | });
      12 | 

      at __tests__/hook-in-empty-nested-describe.test.js:9:3
      at __tests__/hook-in-empty-nested-describe.test.js:8:1

",
  "summary": "Test Suites: 1 failed, 1 total
Tests:       1 passed, 1 total
Snapshots:   0 total
Time:        <<REPLACED>>
Ran all test suites matching /hook-in-empty-nested-describe.test.js/i.
",
}
`;

exports[`multiple hooks in empty describe 1`] = `
Object {
  "rest": "FAIL __tests__/multiple-hooks-in-empty-describe.test.js


  ● Test suite failed to run

    Invalid: beforeEach() may not be used in a describe block containing no tests.

       7 | 
       8 | describe('a block', () => {
    >  9 |   beforeEach(() => {});
         |   ^
      10 |   afterEach(() => {});
      11 |   afterAll(() => {});
      12 |   beforeAll(() => {});

      at __tests__/multiple-hooks-in-empty-describe.test.js:9:3
      at __tests__/multiple-hooks-in-empty-describe.test.js:8:1

  ● Test suite failed to run

    Invalid: afterEach() may not be used in a describe block containing no tests.

       8 | describe('a block', () => {
       9 |   beforeEach(() => {});
    > 10 |   afterEach(() => {});
         |   ^
      11 |   afterAll(() => {});
      12 |   beforeAll(() => {});
      13 | });

      at __tests__/multiple-hooks-in-empty-describe.test.js:10:3
      at __tests__/multiple-hooks-in-empty-describe.test.js:8:1

  ● Test suite failed to run

    Invalid: afterAll() may not be used in a describe block containing no tests.

       9 |   beforeEach(() => {});
      10 |   afterEach(() => {});
    > 11 |   afterAll(() => {});
         |   ^
      12 |   beforeAll(() => {});
      13 | });
      14 | 

      at __tests__/multiple-hooks-in-empty-describe.test.js:11:3
      at __tests__/multiple-hooks-in-empty-describe.test.js:8:1

  ● Test suite failed to run

    Invalid: beforeAll() may not be used in a describe block containing no tests.

      10 |   afterEach(() => {});
      11 |   afterAll(() => {});
    > 12 |   beforeAll(() => {});
         |   ^
      13 | });
      14 | 
      15 | describe('another block with tests', () => {

      at __tests__/multiple-hooks-in-empty-describe.test.js:12:3
      at __tests__/multiple-hooks-in-empty-describe.test.js:8:1

",
  "summary": "Test Suites: 1 failed, 1 total
Tests:       1 passed, 1 total
Snapshots:   0 total
Time:        <<REPLACED>>
Ran all test suites matching /multiple-hooks-in-empty-describe.test.js/i.
",
}
`;
